CMakeLists.txt

```diff
@@ -1,8 +1,42 @@
 cmake_minimum_required(VERSION 3.5)
-project(buildfarm_perf_tests LANGUAGES C)
+project(buildfarm_perf_tests)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
+find_package(Boost REQUIRED COMPONENTS program_options)
+
+include_directories(include)
+
+add_executable(system_metric_collector
```

Comment:
Does this executable work on macOS?

Whatever the case, we should add some logic here so that we don't try to build it on platforms for which it isn't supported.

Reply:
`perf_test` is not compiling in MacOS, should we put some effort in to make this compatible with MacOS?

Reply:
I don't think we need to make it compatible, but we should make an effort to respond appropriately if an attempt is made to compile on macOS or Windows.

There are a lot of examples where we just "skip" the parts of a package that are unsupported on a platform. It would be great to see either that or a CMake error message stating that it isn't supported.